addons/mail/models/mail_link_preview.py

```diff
@@ -38,7 +39,10 @@ def _create_from_message_and_notify(self, message):
         link_previews_by_url = {
             preview.source_url: preview for preview in message.sudo().link_preview_ids
         }
+        ignore_pattern = re.compile(f"{request_url}/(odoo|web)(/|$)") if request_url else None
```

Comment:
Note that `url_root` (always?) contains the trailing `/` so that's probably not working as expected right now (tests aren't realistic in that regard).